### PR TITLE
Various fixes for the Information Box component

### LINKED ISSUE #16271: Inconsistent Quo Information Box implementation
# Problem

While working on issue https://github.com/status-im/status-mobile/issues/16119 to fix the Community Overview screen, I noticed the [Information Box component](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=2950-58462&t=vn4WsmE1ENMSA6xX-0) had lots of subtle inconsistencies.

Upon further inspection we can see the component is incorrectly implemented.

- It reuses the Info Message component ([Figma](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=2950-43503&t=vn4WsmE1ENMSA6xX-0), [code](https://github.com/status-im/status-mobile/blob/350303256b0c5f3a5d6c806482eb3a48c32803b5/src/quo2/components/info/information_box.cljs#L80)), but this is incorrect. It's not how the design system is implemented (they're independent) and it causes incorrect spacing between the icon and text.
- The close icon is not right aligned.
- The main button is incorrectly fixed as `primary`, but it can also be `danger`.

Other problems are more on the technical side because the component was created before the new guidelines were set.

### PR DESCRIPTION
**Pull request by ilmotta**
Fixes https://github.com/status-im/status-mobile/issues/16271

### Summary

This PR fixes UI bugs described in https://github.com/status-im/status-mobile/issues/16271, and it also rewrites the component to follow our guidelines i.e. move code to `style` namespace, fix docstring, add component specs, move to separate quo2 directory, etc.

I'm opening this PR as part of the effort to fix the Community Overview screen.

### Before/after

<table>
  <thead>
    <tr>
      <th>Before</th>
      <th>After</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td></td>
      <td></td>
    </tr>
  </tbody>
</table>

### Steps to test

Go to `Quo2 preview` > `info` > `information-box`

status: ready

### CODE REVIEW

**Review by Parveshdhull — APPROVED** [review]

File: src/quo2/components/info/information_box/style.cljs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                          :text   colors/white}
+           :error        {:bg     colors/danger-50-opa-5
+                          :border colors/danger-50-opa-10
+                          :icon   colors/danger-50
+                          :text   colors/danger-50}
+           :close-button colors/white}})
+
+(defn get-color
+  [key]
+  (get-in themes [(theme/get-theme) key]))
```
Comment (J-Son89) [inline src/quo2/components/info/information_box/style.cljs]: hmm, perhaps you can safe us some future work by using theme as a param and passing it in where it's used? 🤔

File: src/quo2/components/info/information_box/view.cljs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        {:color           (style/get-color-by-type type :icon)
+         :no-color        no-icon-color?
+         :size            (or icon-size 16)
+         :container-style style/icon}]
+       [content
+        {:type            type
+         :button-label    button-label
+         :on-button-press on-button-press
+         :message         message}]
+       (when closable?
```
Comment (J-Son89) [inline src/quo2/components/info/information_box/view.cljs]: can/should an information box be not closable but have an on-close function?

**Review by J-Son89 — APPROVED** [review]
nice work 👍

**Review by rahulpsingh — APPROVED** [review]

File: src/quo2/components/info/information_box/style.cljs (hunk 1 above)
Comment (ilmotta) [reply src/quo2/components/info/information_box/style.cljs]: This is the original code still, but I'll adjust as you suggest :) Thanks

File: src/quo2/components/info/information_box/view.cljs (hunk 2 above)
Comment (ilmotta) [reply src/quo2/components/info/information_box/view.cljs]: This is the original code, but yes, you're right. I'll remove this unnecessary flag from the code. It actually has zero usages. Good suggestion, thanks

Comment (J-Son89) [reply src/quo2/components/info/information_box/view.cljs]: I checked the original pr and there's no comments etc about that prop: 
https://github.com/status-im/status-mobile/pull/13753
So it appears safe to do so 🤞

**Pull request merged**